controllers/state_of_the_world.go

```diff
@@ -45,13 +50,45 @@ func NewPolicyMachineryController(manager ctrlruntime.Manager, client *dynamic.D
 		controller.ManagedBy(manager),
 		controller.WithLogger(logger),
 		controller.WithClient(client),
-		controller.WithRunnable("kuadrant watcher", controller.Watch(&kuadrantv1beta1.Kuadrant{}, kuadrantv1beta1.KuadrantResource, metav1.NamespaceAll)),
-		controller.WithRunnable("dnspolicy watcher", controller.Watch(&kuadrantv1alpha1.DNSPolicy{}, kuadrantv1alpha1.DNSPoliciesResource, metav1.NamespaceAll)),
-		controller.WithRunnable("tlspolicy watcher", controller.Watch(&kuadrantv1alpha1.TLSPolicy{}, kuadrantv1alpha1.TLSPoliciesResource, metav1.NamespaceAll)),
-		controller.WithRunnable("authpolicy watcher", controller.Watch(&kuadrantv1beta2.AuthPolicy{}, kuadrantv1beta2.AuthPoliciesResource, metav1.NamespaceAll)),
-		controller.WithRunnable("ratelimitpolicy watcher", controller.Watch(&kuadrantv1beta2.RateLimitPolicy{}, kuadrantv1beta2.RateLimitPoliciesResource, metav1.NamespaceAll)),
-		controller.WithRunnable("topology configmap watcher", controller.Watch(&corev1.ConfigMap{}, controller.ConfigMapsResource, operatorNamespace, controller.FilterResourcesByLabel[*corev1.ConfigMap](fmt.Sprintf("%s=true", kuadrant.TopologyLabel)))),
-		controller.WithRunnable("limitador watcher", controller.Watch(&limitadorv1alpha1.Limitador{}, kuadrantv1beta1.LimitadorResource, metav1.NamespaceAll)),
+		controller.WithRunnable("kuadrant watcher", controller.Watch(
+			&kuadrantv1beta1.Kuadrant{},
+			kuadrantv1beta1.KuadrantResource,
+			metav1.NamespaceAll,
+			controller.WithPredicates(&ctrlruntimepredicate.TypedGenerationChangedPredicate[*kuadrantv1beta1.Kuadrant]{})),
+		),
+		controller.WithRunnable("dnspolicy watcher", controller.Watch(
+			&kuadrantv1alpha1.DNSPolicy{},
+			kuadrantv1alpha1.DNSPoliciesResource,
+			metav1.NamespaceAll,
+			controller.WithPredicates(&ctrlruntimepredicate.TypedGenerationChangedPredicate[*kuadrantv1alpha1.DNSPolicy]{})),
+		),
+		controller.WithRunnable("tlspolicy watcher", controller.Watch(
+			&kuadrantv1alpha1.TLSPolicy{},
+			kuadrantv1alpha1.TLSPoliciesResource,
+			metav1.NamespaceAll,
+			controller.WithPredicates(&ctrlruntimepredicate.TypedGenerationChangedPredicate[*kuadrantv1alpha1.TLSPolicy]{})),
+		),
+		controller.WithRunnable("authpolicy watcher", controller.Watch(
+			&kuadrantv1beta2.AuthPolicy{},
+			kuadrantv1beta2.AuthPoliciesResource,
+			metav1.NamespaceAll,
+			controller.WithPredicates(&ctrlruntimepredicate.TypedGenerationChangedPredicate[*kuadrantv1beta2.AuthPolicy]{})),
+		),
+		controller.WithRunnable("ratelimitpolicy watcher", controller.Watch(
+			&kuadrantv1beta2.RateLimitPolicy{},
+			kuadrantv1beta2.RateLimitPoliciesResource,
+			metav1.NamespaceAll),
```

Comment:
Any reason why we wouldn't have `TypedGenerationChangedPredicate` for RateLimitPolicy too?

Reply:
Hmm, no, I think I just missed this, will update 👍